### ISSUE TITLE
Missing Python dependency "six" in "bench init"

### ISSUE THREAD

**Issue by danielmorlock** [frappe/bench#425]
The following command fails on a fresh install:

```bash
$ sudo -u frappe HOME=/home/frappe bench init /home/frappe/frappe-bench --frappe-path /tmp/frappe-repo
root@12ba2cfbdf6f:/home/frappe# sudo -u frappe HOME=/home/frappe bench init /home/frappe/frappe-bench --frappe-path /tmp/frappe-repo --verbose
*** setup_env...
*** virtualenv
Already using interpreter /usr/bin/python
*** upgrade pip
Requirement already up-to-date: pip in ./env/lib/python2.7/site-packages
*** install wheel
Requirement already satisfied: wheel in ./env/lib/python2.7/site-packages
*** install pdfkit
Obtaining pdfkit from git+https://github.com/frappe/python-pdfkit.git#egg=pdfkit
  Cloning https://github.com/frappe/python-pdfkit.git to ./env/src/pdfkit
Installing collected packages: pdfkit
  Running setup.py develop for pdfkit
    Complete output from command /home/frappe/frappe-bench/env/bin/python -c "import setuptools, tokenize;__file__='/home/frappe/frappe-bench/env/src/pdfkit/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" develop --no-deps:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/home/frappe/frappe-bench/env/src/pdfkit/setup.py", line 51, in <module>
        'Topic :: Utilities'
      File "/usr/lib/python2.7/distutils/core.py", line 137, in setup
        ok = dist.parse_command_line()
      File "/home/frappe/frappe-bench/env/local/lib/python2.7/site-packages/setuptools/dist.py", line 363, in parse_command_line
        result = _Distribution.parse_command_line(self)
      File "/usr/lib/python2.7/distutils/dist.py", line 467, in parse_command_line
        args = self._parse_command_opts(parser, args)
      File "/home/frappe/frappe-bench/env/local/lib/python2.7/site-packages/setuptools/dist.py", line 674, in _parse_command_opts
        nargs = _Distribution._parse_command_opts(self, parser, args)
      File "/usr/lib/python2.7/distutils/dist.py", line 523, in _parse_command_opts
        cmd_class = self.get_command_class(command)
      File "/home/frappe/frappe-bench/env/local/lib/python2.7/site-packages/setuptools/dist.py", line 495, in get_command_class
        self.cmdclass[command] = cmdclass = ep.load()
      File "/home/frappe/frappe-bench/env/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2303, in load
        return self.resolve()
      File "/home/frappe/frappe-bench/env/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2309, in resolve
        module = __import__(self.module_name, fromlist=['__name__'], level=0)
      File "/home/frappe/frappe-bench/env/local/lib/python2.7/site-packages/setuptools/command/develop.py", line 11, in <module>
        from setuptools.command.easy_install import easy_install
      File "/home/frappe/frappe-bench/env/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 49, in <module>
        from setuptools.py27compat import rmtree_safe
      File "/home/frappe/frappe-bench/env/local/lib/python2.7/site-packages/setuptools/py27compat.py", line 7, in <module>
        import six
    ImportError: No module named six
    
    ----------------------------------------
Command "/home/frappe/frappe-bench/env/bin/python -c "import setuptools, tokenize;__file__='/home/frappe/frappe-bench/env/src/pdfkit/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" develop --no-deps" failed with error code 1 in /home/frappe/frappe-bench/env/src/pdfkit/
Traceback (most recent call last):
  File "/usr/local/bin/bench", line 9, in <module>
    load_entry_point('bench', 'console_scripts', 'bench')()
  File "/opt/bench-4.1/bench/cli.py", line 40, in cli
    bench_command()
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/opt/bench-4.1/bench/commands/make.py", line 17, in init
    no_auto_update=no_auto_update, frappe_path=frappe_path, frappe_branch=frappe_branch, verbose=verbose)
  File "/opt/bench-4.1/bench/utils.py", line 46, in init
    setup_env(bench_path=path)
  File "/opt/bench-4.1/bench/utils.py", line 105, in setup_env
    exec_cmd('./env/bin/pip install -e git+https://github.com/frappe/python-pdfkit.git#egg=pdfkit', cwd=bench_path)
  File "/opt/bench-4.1/bench/utils.py", line 94, in exec_cmd
    raise CommandFailedError(cmd)
```

Note that I added some debug output and invoke `pip install` without `--quiet` so see what is going on. Obviously, the module `six` is missing in the virtualenv. After manually installing `six`, I can also install `python-pdfkit`:

```bash
root@12ba2cfbdf6f:/home/frappe# source frappe-bench/env/bin/activate
(env) root@12ba2cfbdf6f:/home/frappe# pip install six
Collecting six
  Using cached six-1.10.0-py2.py3-none-any.whl
Installing collected packages: six
Successfully installed six-1.10.0
(env) root@12ba2cfbdf6f:/home/frappe# pip install -e "git+https://github.com/frappe/python-pdfkit.git#egg=pdfkit"
Obtaining pdfkit from git+https://github.com/frappe/python-pdfkit.git#egg=pdfkit
  Updating ./frappe-bench/env/src/pdfkit clone
Installing collected packages: pdfkit
  Running setup.py develop for pdfkit
Successfully installed pdfkit
(env) root@12ba2cfbdf6f:/home/frappe# 
```

Maybe one should add the requirement to `setup.py` of python-pdfkit (https://github.com/frappe/python-pdfkit/blob/master/setup.py#L29)?

**Comment by danielmorlock** [comment]
Workaround:

```bash
sed -s 's#pip -q install wheel#pip -q install six wheel#' -i /path/to/bench-4.1/bench/utils.py
```

**Comment by achillesrasquinha** [comment]
Fixed - #435
